src/transaction.rs

```diff
@@ -92,6 +92,8 @@ pub struct Transaction {
 
     /// The number of tokens paid for processing and storage of this transaction.
     pub fee: i64,
+    /// Optional user data to be stored in the account
+    pub user_data: Option<Vec<u8>>,
```

Comment:
I'd prefer we not add optional fields to `Transaction`. Instead, `Instruction` should be the top-level entrypoint for consideration. In this case, you may consider the `Budget` enum, and see this as an alternative to the `Pay` constructor. Add `Allocate`?

Reply:
@garious `user_data = serialize(Instruction{...});` . So its not optional.  the Most basic spend will need to serialize the amount of tokens to move. Its just opaque data for the contract.